src/main/java/org/truffleruby/language/supercall/LookupSuperMethodNode.java

```diff
@@ -44,15 +49,15 @@
     protected InternalMethod lookupSuperMethodCachedDynamicObject(VirtualFrame frame, DynamicObject self,
```

Comment:
Note: the guards would need to check that the caller frame/parent's DeclarationContext is the same, as it could change if the Module is included in multiple refinements (I wonder if that actually works on MRI).